src/Compiler/Checking/TailCallChecks.fs

```diff
@@ -756,7 +756,25 @@ let CheckModuleBinding cenv (isRec: bool) (TBind _ as bind) =
                 | Expr.Lambda(bodyExpr = bodyExpr) -> checkTailCall insideSubBindingOrTry bodyExpr
                 | Expr.DebugPoint(_debugPointAtLeafExpr, expr) -> checkTailCall insideSubBindingOrTry expr
                 | Expr.Let(binding = binding; bodyExpr = bodyExpr) ->
-                    checkTailCall true binding.Expr
+                    let isAsyncContinuation =
+                        match bodyExpr with
+                        | Expr.App(funcExpr = Expr.Val(valRef = valRef)) ->
```

Comment:
Not super sure about this. A simpler test might be:
```
when
                            valRef.DisplayNameCore = "MakeAsync"
                            && valRef.HasDeclaringEntity
                            && valRef.DeclaringEntity.LogicalName = "AsyncPrimitives"
```

But that might be too much cowboy-style :cowboy_hat_face:

Reply:
I don't think we should hardcode implementation specifics (from FSharp.Core) into compiler checking.

If there is no other way to detect this as of now, I would have another proposal:
Make it a reusable component, e.g. an attribute place on the decleration side (in this case, within AsyncPrimitives implementation) to make it less hardcoded.